### ISSUE TITLE
Reduce number of flags

### ISSUE THREAD

**Issue by marcadella** [geco-bern/rsofun#242]
Taking the example of `do_spinup`, it is possible to simplify the interface by removing this flag and using `spinupyears=0` instead. It is more compact, less prone to user errors. It is possible to do the same with other flags.

**Comment by marcadella** [comment]
@stineb Is this desirable?

**Comment by stineb** [comment]
Indeed, using `spinupyears` and interpreting `if (spinupyears == 0) do_spinup`, we can avoid the flag `do_spinup`.

**Comment by marcadella** [comment]
Should I do this before submitting to CRAN?

**Comment by stineb** [comment]
Not necessary.

**Comment by fabern** [comment]
> Indeed, using `spinupyears` and interpreting `if (spinupyears == 0) do_spinup`, we can avoid the flag `do_spinup`.

Just note, that this should then idally also be propagated to the interfaces of the `p-model`. 
As stated, I'd be in favor of streamlining those, especially the additional `use_x`-flags that are in the `phydro` branch.
